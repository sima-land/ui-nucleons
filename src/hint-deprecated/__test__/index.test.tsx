import React from 'react';
import { mount } from 'enzyme';
import { Hint } from '..';

describe('<Hint />', () => {
  for (const direction of ['top', 'right', 'bottom', 'left'] as const) {
    it('should renders correctly', () => {
      const wrapper = mount(
        <Hint className='additional-class' direction={direction}>
          Hello, world!
        </Hint>,
      );

      expect(wrapper).toMatchSnapshot();
    });
  }
});
