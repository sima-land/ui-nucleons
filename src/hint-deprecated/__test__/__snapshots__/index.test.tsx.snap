// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hint /> should renders correctly 1`] = `
<ForwardRef
  className="additional-class"
  direction="top"
>
  <div
    className="root additional-class top"
    data-testid="hint"
  >
    <div
      className="body shadow-z4"
    >
      Hello, world!
    </div>
  </div>
</ForwardRef>
`;

exports[`<Hint /> should renders correctly 2`] = `
<ForwardRef
  className="additional-class"
  direction="right"
>
  <div
    className="root additional-class right"
    data-testid="hint"
  >
    <div
      className="body shadow-z4"
    >
      Hello, world!
    </div>
  </div>
</ForwardRef>
`;

exports[`<Hint /> should renders correctly 3`] = `
<ForwardRef
  className="additional-class"
  direction="bottom"
>
  <div
    className="root additional-class bottom"
    data-testid="hint"
  >
    <div
      className="body shadow-z4"
    >
      Hello, world!
    </div>
  </div>
</ForwardRef>
`;

exports[`<Hint /> should renders correctly 4`] = `
<ForwardRef
  className="additional-class"
  direction="left"
>
  <div
    className="root additional-class left"
    data-testid="hint"
  >
    <div
      className="body shadow-z4"
    >
      Hello, world!
    </div>
  </div>
</ForwardRef>
`;
